File: 0172-factorial-trailing-zeroes/0172-factorial-trailing-zeroes.java
class Solution {
  public int trailingZeroes(int n) {
    List<Integer> powers = new ArrayList<Integer>();
    int b = 5;
    while (b <= 10000) {
      powers.add(b);
      b *= 5;
    }
    return powers.stream()
      .reduce(0, (sum, base) -> sum + n / base);
  }
}

File: 0172-factorial-trailing-zeroes/0172-factorial-trailing-zeroes.js
/**
 * @param {number} n
 * @return {number}
 */
var trailingZeroes = function(n) {
  const powers = [];
  let b = 5;
  while (b < 10000) {
    powers.push(b);
    b *= 5;
  }
  return powers.reduce((sum, base) => sum + Math.floor(n / base), 0);
};